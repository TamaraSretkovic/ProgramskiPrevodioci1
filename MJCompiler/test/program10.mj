//	Semantika test 
//  - expression print mora biti int, char ili bool

program test302
    int d;
    const char c = 'a';
{
    void func() {

    }

	void main() {
		print(func);
        print(c);
	}
}
