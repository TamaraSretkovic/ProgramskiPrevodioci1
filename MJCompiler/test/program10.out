INFO  01:15:55,941 - Compiling source file: /home/tamarica/Documents/tamara/velika skola/8 semestar/PP1/projekat/projecat_2019_2020/MJCompiler/test/program10.mj
INFO  01:15:55,964 - #2 program
INFO  01:15:55,964 - #46 test302
INFO  01:15:55,964 - #46 int
INFO  01:15:55,965 - #46 d
INFO  01:15:55,965 - #5 ;
INFO  01:15:55,966 - #7 const
INFO  01:15:55,966 - #46 char
INFO  01:15:55,966 - #46 c
INFO  01:15:55,966 - #23 =
INFO  01:15:55,966 - #48 a
INFO  01:15:55,967 - #5 ;
INFO  01:15:55,967 - #3 {
INFO  01:15:55,967 - #10 void
INFO  01:15:55,968 - #46 func
INFO  01:15:55,968 - #12 (
INFO  01:15:55,968 - #11 )
INFO  01:15:55,969 - #3 {
INFO  01:15:55,969 - #4 }
INFO  01:15:55,969 - #10 void
INFO  01:15:55,970 - #46 main
INFO  01:15:55,970 - #12 (
INFO  01:15:55,970 - #11 )
INFO  01:15:55,971 - #3 {
INFO  01:15:55,971 - #16 print
INFO  01:15:55,971 - #12 (
INFO  01:15:55,971 - #46 func
INFO  01:15:55,971 - #11 )
INFO  01:15:55,971 - #5 ;
INFO  01:15:55,972 - #16 print
INFO  01:15:55,972 - #12 (
INFO  01:15:55,972 - #46 c
INFO  01:15:55,972 - #11 )
INFO  01:15:55,972 - #5 ;
INFO  01:15:55,972 - #4 }
INFO  01:15:55,972 - #4 }
INFO  01:15:55,977 - ==================SINTAKSNO STABLO==================
INFO  01:15:55,977 - Program(
  ProgName(
   test302
  ) [ProgName]
  DeclarationListDeclaration(
    DeclarationListDeclaration(
      NoDeclarationList(
      ) [NoDeclarationList]
      DeclarationListVar(
        VarDeclarationV(
          VarType(
            Type(
             int
            ) [Type]
          ) [VarType]
          VarDecl(
            Ident(
             d
            ) [Ident]
          ) [VarDecl]
        ) [VarDeclarationV]
      ) [DeclarationListVar]
    ) [DeclarationListDeclaration]
    DeclarationListConst(
      ConstDeclarationTypeListSemi(
        VarType(
          Type(
           char
          ) [Type]
        ) [VarType]
        ConstDecl(
          ConstDeclarationActual(
           c
            AssignOp(
            ) [AssignOp]
            CharConst(
             a
            ) [CharConst]
          ) [ConstDeclarationActual]
        ) [ConstDecl]
      ) [ConstDeclarationTypeListSemi]
    ) [DeclarationListConst]
  ) [DeclarationListDeclaration]
  MethodDeclarationListMethod(
    MethodDeclarationListMethod(
      NoMethodDeclarationList(
      ) [NoMethodDeclarationList]
      VoidMethodDeclaration(
        MethodVoidName(
         func
        ) [MethodVoidName]
        MethodDeclarationActual(
          NoFormalParamsList(
          ) [NoFormalParamsList]
          NoDeclarationList(
          ) [NoDeclarationList]
          NoStatementList(
          ) [NoStatementList]
        ) [MethodDeclarationActual]
      ) [VoidMethodDeclaration]
    ) [MethodDeclarationListMethod]
    VoidMethodDeclaration(
      MethodVoidName(
       main
      ) [MethodVoidName]
      MethodDeclarationActual(
        NoFormalParamsList(
        ) [NoFormalParamsList]
        NoDeclarationList(
        ) [NoDeclarationList]
        StatementListStatement(
          StatementListStatement(
            NoStatementList(
            ) [NoStatementList]
            PrintExpressionStatement(
              ExpressionTermDef(
                OnlyTerm(
                  TermFactor(
                    FactorDesignator(
                      DesignatorOnly(
                       func
                      ) [DesignatorOnly]
                    ) [FactorDesignator]
                  ) [TermFactor]
                ) [OnlyTerm]
              ) [ExpressionTermDef]
            ) [PrintExpressionStatement]
          ) [StatementListStatement]
          PrintExpressionStatement(
            ExpressionTermDef(
              OnlyTerm(
                TermFactor(
                  FactorDesignator(
                    DesignatorOnly(
                     c
                    ) [DesignatorOnly]
                  ) [FactorDesignator]
                ) [TermFactor]
              ) [OnlyTerm]
            ) [ExpressionTermDef]
          ) [PrintExpressionStatement]
        ) [StatementListStatement]
      ) [MethodDeclarationActual]
    ) [VoidMethodDeclaration]
  ) [MethodDeclarationListMethod]
) [Program]
INFO  01:15:55,978 - ==================SEMANTICKA OBRADA==================
INFO  01:15:55,980 - Deklarisana promenljiva d(Var d: int, 0, 0 ) na liniji 5
INFO  01:15:55,980 - Deklarisana konstanta c(Con c: char, 97, 0 ) na liniji 6
INFO  01:15:55,980 - Obradjuje se funkcija func(Meth func: none, 0, 0 ) na liniji 8
INFO  01:15:55,980 - Obradjuje se funkcija main(Meth main: none, 0, 0 ) na liniji 12
INFO  01:15:55,981 - Koriscenje  func(Meth func: none, 0, 0 ) na liniji 13
ERROR 01:15:55,981 - Greska na liniji 13 : izraz PrintExpressionStatement(
  ExpressionTermDef(
    OnlyTerm(
      TermFactor(
        FactorDesignator(
          DesignatorOnly(
           func
          ) [DesignatorOnly]
        ) [FactorDesignator]
      ) [TermFactor]
    ) [OnlyTerm]
  ) [ExpressionTermDef]
) [PrintExpressionStatement] nije odgovarajuceg tipa!
INFO  01:15:55,981 - Koriscenje  c(Con c: char, 97, 0 ) na liniji 14
INFO  01:15:55,981 - Pronadjen poziv funkcije print na liniji 14
INFO  01:15:55,981 - ==================SEMANTICKA ANALIZA==============
INFO  01:15:55,982 - 2  methods in the program
INFO  01:15:55,982 - 1  global variables
INFO  01:15:55,982 - 1  global constants
INFO  01:15:55,982 - 0  global arrays
INFO  01:15:55,982 - 0  local variables in main
INFO  01:15:55,982 - 2  statements in main
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: bool, 0, 1 
Prog test302: notype, 0, 1 
   Var d: int, 0, 0 
   Con c: char, 97, 0 
   Meth func: notype, 0, 0 

   Meth main: notype, 0, 0 



ERROR 01:15:55,983 - Program nije semanticki ispravan!
